### ISSUE TITLE
Access output of processes without named output channel

### ISSUE THREAD

**Issue by byb121** [askimed/nf-test#55]
Nf-test provides a very convenient way to access named output channels of a Nextflow process, e.g.: `process.out.my_outCh`,  but many of our Nextflow processes do not create a named output channel, e.g.: not using `emit` the script. I wonder if there Is any method to access these outputs for assertions conveniently?

**Comment by nvnieuwk** [comment]
Hi, does `process.out.get(0)` not work?

**Comment by byb121** [comment]
Hi @nvnieuwk,

I've tried it. It does not solve the issue. Without a named channel, `process.out` is an empty map.

With a little change on one of the existing examples, we can prove it:

```
$ cat example/say-hello.nf
process SAY_HELLO {

  input:
    val name
    val number

  output:
     tuple val(number), val(name)
 
  script:
  """
  echo "Hi there! I am ${name} number ${number}" > "${number}_${name}.txt"
  """
}

$ cat example/say-hello.nf.test 
nextflow_process {

  name "Test process SAY_HELLO"

  script "example/say-hello.nf"
  process "SAY_HELLO"

  test("Running with positive number should succeed") {

    when {
      process {
        """
        input[0] = "sebastian"
        input[1] = Channel.of(
          [['2':'test', 'test2':['test3': 'test4']]],
          [['3':'test', 'test4':['test5': 'test6']]]
        )
        """
      }
    }

    then {
      assert process.success
      assert process.out.get(0) == [
        [[['2':'test', 'test2':['test3': 'test4']]], "sebastian"],
        [[['3':'test', 'test4':['test5': 'test6']]], "sebastian"]
      ]
    }
  }
}

$ nf-test test example/say-hello.nf.test 

🚀 nf-test 0.7.1
https://code.askimed.com/nf-test
(c) 2021 - 2022 Lukas Forer and Sebastian Schoenherr

Warning: This pipeline has no nf-test config file.

Test process SAY_HELLO

  Test [24e99c52] 'Running with positive number should succeed' [:]
FAILED (4.262s)

  Assertion failed: 
  
  assert process.out.get(0) == [ [[['2':'test', 'test2':['test3': 'test4']]], "sebastian"], [[['3':'test', 'test4':['test5': 'test6']]], "sebastian"] ]
         |       |   |      |
         |       [:] null   false
         SAY_HELLO
  
  Nextflow stdout:
  
  Nextflow stderr:

FAILURE: Executed 1 tests in 4.273s (1 failed)

```

**Comment by lukfor** [comment]
I have a local branch that solves this issue. I can push it tomorrow.

**Comment by lukfor** [comment]
I pushed the branch (junit testcases are not yet implemented):

https://github.com/askimed/nf-test/tree/features/unnamed-output-channels

Could you give it a try?

This should work:

```
assert process.out.get(0) == [
        [[['2':'test', 'test2':['test3': 'test4']]], "sebastian"],
        [[['3':'test', 'test4':['test5': 'test6']]], "sebastian"]
      ]
```
This should also work:
```
assert process.out[0] == [
        [[['2':'test', 'test2':['test3': 'test4']]], "sebastian"],
        [[['3':'test', 'test4':['test5': 'test6']]], "sebastian"]
      ]
```

**Comment by byb121** [comment]
Hi @lukfor,

That works great in my hand. Thanks for the quick fix. It'll be great if we can follow Nextflow's way to handle process/workflow's output, e.g.: `process.out` for unnamed output, `process.out[0], process.out[1]` for first and second channel and `process.out.outputCh` for a named output channel. This means developers can quickly move their Nextflow code into test cases with minimum changes.

**Comment by lukfor** [comment]
Yes, that makes totally sense.

We support it already or have I missed something?

**Comment by byb121** [comment]
Apologies, I played output channels in nf-test a bit more and yes,  it's almost identical to a channel in Nextflow. The only difference I found is that in Nextflow `process.out` can return the channel content if a process only outputs a single element, either named or unnamed, while in nf-test, we'll have to use `get(0)` or use the channel name. However, I suppose it'll require quite a bit of work to implement this in nf-test.
